ert/storage/_storage.py

```diff
@@ -71,20 +69,33 @@ def __init__(self, name: str, storage_url: str, iens: Optional[int] = None):
         self._name: str = name
         self._uri = f"{storage_url}/{name}"
         self._real_id: Optional[int] = iens
-        if self._real_id is not None:
-            self._uri = f"{self._uri}?realization_index={self._real_id}"
+
+    @property
+    def uri(self) -> str:
+        assert self.is_transmitted()
+        return self._uri
+
+    @property
+    def record_type(self) -> Optional[ert.data.RecordType]:
+        assert self.is_transmitted()
+        return self._record_type
+
+    def set_transmitted(self, uri: str, record_type: ert.data.RecordType) -> None:
+        self._set_transmitted_state(uri, record_type)
```

Comment:
I'd really like to hide this `_set_transmitted_state` and rather implement a `RecordTransmitter.from_transmitted()` which is general and to be implemented for all transmitter types. Because if this is introduced as-is, the `RecordTransmitter` interface is obsolete.

Reply:
Good point!

Reply:
After thinking about it a bit more I realized I don't really understand what you would want you would want `RecordTransmitter.from_transmitted()` to return. A general RecordTransmitter ?  Then the implementations for each of the transmitter types would return an instance of that Transmitter type that is filled in with name, uri and record type and the state is set to  `RecordTransmitterState.transmitted`. But if that is the case would it not make more sense to just move the implementation of `set_transmitted` into the `RecordTrasmitter` interface.  

Also, I am not sure how are you thinking the usage would go for an "already" transmitted inmemory transmitter. Not really sure how to wrap my brain around that.

Reply:
Mainly I wanted to keep the `RecordTransmitter` interface relevant. If the interface is turbulent (which they often are), it may be indicative of a problematic abstraction.

Also, I wanted to protect the "transmitted" state. I see it as a side-effect from doing real work (like actually transmitting stuff), and not really something that should be generally available to mutate.

Does that make sense?

Reply:
It's fine that a transmitter subclass raises a NotImplementedError.